### ISSUE TITLE
new link/path busted for any refocus that isn't trivial

### ISSUE THREAD

**Issue by dustingetz** [hyperfiddle/hyperfiddle#839]
This function is broken
```
(defn refocus "todo unify with refocus'"
  [ctx path]
  {:pre [ctx] :post [(contains? % :hypercrud.browser/eav)]}
  (let [current-path (:hypercrud.browser/path ctx)
        common-ancestor-path (ancestry-common current-path path)
        unwind-offset (- (count current-path) (count common-ancestor-path))
        common-ancestor-ctx ((apply comp (repeat unwind-offset :hypercrud.browser/parent)) ctx)]
    (focus common-ancestor-ctx (ancestry-divergence path current-path))))
```

#843

**Comment by dustingetz** [comment]
From data mode perspective, we traverse the resultset, checking the :hypercrud.browser/eav against the link/a at each step, which is trivial to lookup "what links are good here"

From the view perspective, the view names a specific single link and from that link we need to reconstruct the link's formula context. For anchors and buttons, you are only naming a link from inside the correct context, or one step away from the correct context (the row of the table is known for example). Is this also true for iframes?

Iframes are often relocated around, e.g. for options. 
* !link(:genders) can be used as options anywhere on the page, since its dependencies are always filled.
* !link(:shirt-size) is legal from anywhere downtree of it's :link/a; so pop until eav matches :shirt-size :link/a

**Comment by dustingetz** [comment]
I think this is fixed in #854
